server/channels/jobs/mobile_session_metadata/worker.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package mobile_session_metadata
+
+import (
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"github.com/mattermost/mattermost/server/v8/channels/jobs"
+	"github.com/mattermost/mattermost/server/v8/channels/store"
+	"github.com/mattermost/mattermost/server/v8/einterfaces"
+)
+
+func MakeWorker(jobServer *jobs.JobServer, store store.Store, getMetrics func() einterfaces.MetricsInterface) *jobs.SimpleWorker {
+	const workerName = "MobileSessionMetadata"
+
+	isEnabled := func(cfg *model.Config) bool {
+		return *cfg.MetricsSettings.EnableClientMetrics
+	}
+	execute := func(logger mlog.LoggerIFace, job *model.Job) error {
+		defer jobServer.HandleJobPanic(logger, job)
+
+		metrics := getMetrics()
+		if metrics == nil {
+			return nil
+		}
+
+		versions, err := store.Session().GetMobileSessionMetadata()
+		if err != nil {
+			return err
+		}
+
```

Comment:
Do we want to log any errors in this cases? Seems that the parent doesn't look at the returned `err`

Reply:
It is not clear where the log is created, but the "parent" logs the error.

The idea is that this will create a worker with this `execute` function. The worker will be passed around, and eventually, the method `Run` of `SimpleWorker` will be called, which eventually calls `DoJob`, and this is where `execute` is really called (`channels/jobs/base_workers.go` line 94). There the error is logged.

Reply:
Thanks for clarifying, I'm not too familiar with how the jobs are setup 👍